c7n/resources/s3.py

```diff
@@ -1416,35 +1417,26 @@ def process(self, buckets, event=None):
 
     def process_bucket(self, bucket):
         s3 = bucket_client(local_session(self.manager.session_factory), bucket)
-        enabled = self.data.get('enabled', False)
-        scope = self.data.get('scope', 'Any')
+        config = {}
         try:
             config = s3.get_public_access_block(
                 Bucket=bucket['Name'])['PublicAccessBlockConfiguration']
         except ClientError as e:
             if e.response['Error']['Code'] == 'NoSuchPublicAccessBlockConfiguration':
-                # Set config to none because NoSuchPublicAccessBlockConfiguration error
-                # was returned. This is important later because it is symbolic of
-                # all blocks being set to false. See if/else in matches_filter func
-                config = None
+                # When there is a None config, set all to false to be symbolic
+                for key in self.keys:
+                    config[key] = False
             else:
                 raise
-        if self.matches_filter(config, enabled, scope):
-            return {"Name": bucket['Name'], "publicblocks": config}
-
-    def matches_filter(self, config, enabled, scope):
-        if config:
-            if scope == 'All':
-                return all(config.values()) if enabled else not any(config.values())
-            elif scope == 'Any':
-                return any(config.values()) if enabled else not all(config.values())
-            else:
-                return config[scope] if enabled else not config[scope]
+        if self.matches_filter(config):
+            return {"Name": bucket['Name'], "PublicAccessBlockConfiguration": config}
+
+    def matches_filter(self, config):
+        key_set = [key for key in self.keys if type(self.data.get(key)) is bool]
+        if key_set:
+            return all([self.data.get(key) is config[key] for key in key_set])
         else:
-            # When there is a null/none config, treat it as meaning no public blocks
-            # return False if checking for enabled=True because all are false
-            # return True if checking for enabled=False
-            return False if enabled else True
+            return not all(config.values())
```

Comment:
we want this to return true if any of the values are set to false, it only does that now if all of them are set to false.

```
>>> a = {'a': True, 'b': False}
>>> all(a.values())
False
>>> not all(a.values())
True
>>> a = {'a': True, 'b': True}
>>> not all(a.values())
False
>>> a = {'a': False, 'b': False}
>>> not all(a.values())
True
```

Reply:
So in this case I think it still does that. all will only return true when all of the items/values are true, therefore checking not all will always yield True when one or more of the items/values are false